src/Symfony/Component/Process/Process.php

```diff
@@ -242,11 +242,19 @@ public function start($callback = null)
             if (!isset($this->options['bypass_shell'])) {
                 $this->options['bypass_shell'] = true;
             }
-        }
+        } elseif (!$this->useFileHandles && $this->enhanceSigchildCompatibility && $this->isSigchildEnabled()) {
+            // last exit code is output on the fourth pipe and caught to work around --enable-sigchild
+            $descriptors[3] = array('pipe', 'w');
+
+            $commandline = '';
+            foreach (self::$posixSignals as $s) {
+                $commandline .= "trap 'echo s$s >&3' $s;";
+            }
 
-        $ptsWorkaround = null;
+            // See https://unix.stackexchange.com/questions/71205/background-process-pipe-input
+            $commandline .= '{ ('.$this->commandline.') <&3 3<&- 3>/dev/null & } 3<&0;';
+            $commandline .= 'pid=$!; echo $pid >&3; wait $pid; code=$?; echo x$code >&3; exit $code';
```

Comment:
the old code had a ``p`` prefix before the pid, which is not there anymore. Does it still work fine ?